src/main/java/com/datalogics/pdf/security/LunaHsmManager.java

```diff
@@ -71,8 +71,10 @@ public void hsmLogin(final HsmLoginParameters parms) {
 
         if (slotManager.isLoggedIn()) {
             state = ConnectionState.CONNECTED;
+            loadKeyStore();
+        } else {
+            throw new IllegalArgumentException("Could not log into the Luna HSM");
```

Comment:
Should the Exception thrown at line 76 be a IllegalArgumentException or a IllegalStateException seeing that the manager not connecting  could be for reasons other than the login parameters passed to this method